### PR TITLE
ddl: fix after truncate table 'pre-split-regions' does't work

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

```
mysql> create table t (id int) shard_row_id_bits=2 pre_split_regions=2;
Query OK, 0 rows affected (0.01 sec)

mysql> show table t regions;
+-----------+-----------------------------+-----------------------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| REGION_ID | START_KEY                   | END_KEY                     | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
+-----------+-----------------------------+-----------------------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
|         3 |                             | t_179_r_2305843009213693952 |         2 |               1 | 2     |          0 |             0 |          0 |                    0 |                0 |
|         4 | t_179_r_2305843009213693952 | t_179_r_4611686018427387904 |         5 |               1 | 5     |          0 |             0 |          0 |                    0 |                0 |
|         6 | t_179_r_4611686018427387904 | t_179_r_6917529027641081856 |         7 |               1 | 7     |          0 |             0 |          0 |                    0 |                0 |
|         8 | t_179_r_6917529027641081856 |                             |         9 |               1 | 9     |          0 |             0 |          0 |                    0 |                0 |
+-----------+-----------------------------+-----------------------------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
4 rows in set (0.00 sec)

mysql> truncate table t;
Query OK, 0 rows affected (0.02 sec)

mysql> show table t regions;
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
| REGION_ID | START_KEY | END_KEY | LEADER_ID | LEADER_STORE_ID | PEERS | SCATTERING | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE(MB) | APPROXIMATE_KEYS |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
|         3 |           |         |         2 |               1 | 2     |          0 |             0 |          0 |                    0 |                0 |
+-----------+-----------+---------+-----------+-----------------+-------+------------+---------------+------------+----------------------+------------------+
1 row in set (0.00 sec)
```

After truncate table, the pre-split-region doesn't work.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Several lines of code.

How it Works:

Pre-split table after `truncate table` if the table has 'pre-split-regions' property.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @bb7133

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3753,6 +3753,14 @@ func (d *ddl) TruncateTable(ctx sessionctx.Context, ti ast.Ident) error {
 		}
 		return errors.Trace(err)
 	}
+	oldTblInfo := tb.Meta()
+	if oldTblInfo.PreSplitRegions > 0 {
+		if _, tb, err := d.getSchemaAndTableByIdent(ctx, ti); err == nil {
+			d.preSplitAndScatter(ctx, tb.Meta())
+		}
+
```
Comment (crazycs520) [inline ddl/ddl_api.go:3761]: remove this blank tline.

**Comment by tiancaiamao** [comment]
/run-all-tests
@lysu

**Comment by tiancaiamao** [comment]
PTAL @imtbkcat

**Comment by tiancaiamao** [comment]
PTAL @crazycs520

**Pull request merged**